cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go

```diff
@@ -39,20 +42,35 @@ const (
 
 // UploadConfiguration saves the InitConfiguration used for later reference (when upgrading for instance)
 func UploadConfiguration(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error {
-
 	fmt.Printf("[uploadconfig] storing the configuration used in ConfigMap %q in the %q Namespace\n", kubeadmconstants.InitConfigurationConfigMap, metav1.NamespaceSystem)
 
+	// Prepare the ClusterConfiguration for upload
+	// The components store their config in their own ConfigMaps, then reset the .ComponentConfig struct;
 	// We don't want to mutate the cfg itself, so create a copy of it using .DeepCopy of it first
-	clusterConfigToUpload := cfg.ClusterConfiguration.DeepCopy()
-	// TODO: Reset the .ComponentConfig struct like this:
-	// cfgToUpload.ComponentConfigs = kubeadmapi.ComponentConfigs{}
-	// in order to not upload any other components' config to the kubeadm-config
-	// ConfigMap. The components store their config in their own ConfigMaps.
-	// Before this line can be uncommented util/config.loadConfigurationBytes()
-	// needs to support reading the different components' ConfigMaps first.
-
-	// Marshal the object into YAML
-	cfgYaml, err := configutil.MarshalKubeadmConfigObject(clusterConfigToUpload)
+	clusterConfigurationToUpload := cfg.ClusterConfiguration.DeepCopy()
+	clusterConfigurationToUpload.ComponentConfigs = kubeadmapi.ComponentConfigs{}
+
+	// Marshal the ClusterConfiguration into YAML
+	clusterConfigurationYaml, err := configutil.MarshalKubeadmConfigObject(clusterConfigurationToUpload)
+	if err != nil {
+		return err
+	}
+
+	// Prepare the ClusterStatus for upload
+	// Gets the current cluster status
+	clusterStatus, err := getClusterStatus(client)
```

Comment:
I think we are ok for now, but we should add a //TODO use configmap locks on this object on the get before the update.  We may even bug fix that after the freeze 

/cc @liztio

Reply:
done